src/deluge/model/clip/instrument_clip.cpp

```diff
@@ -2458,7 +2399,38 @@ void InstrumentClip::writeDataToFile(Serializer& writer, Song* song) {
 		writer.writeAttribute("noteMode", (char*)arpNoteModeToString(arpSettings.noteMode));
 		writer.writeAttribute("octaveMode", (char*)arpOctaveModeToString(arpSettings.octaveMode));
 		writer.writeAttribute("mpeVelocity", (char*)arpMpeModSourceToString(arpSettings.mpeVelocity));
-
+		writer.writeAttribute("stemRepeat", arpSettings.numStepRepeats);
+		writer.writeAttribute("randomizerLock", arpSettings.randomizerLock);
+
+		// Write locked params
+		// Note probability
+		writer.writeAttribute("lastLockedNoteProb", arpSettings.lastLockedNoteProbabilityParameterValue);
+		writer.writeAttributeHexBytes("lockedNoteProbArray", (uint8_t*)arpSettings.lockedNoteProbabilityValues,
+		                              RANDOMIZER_LOCK_MAX_SAVED_VALUES);
+		// Bass probability
+		writer.writeAttribute("lastLockedBassProb", arpSettings.lastLockedBassProbabilityParameterValue);
+		writer.writeAttributeHexBytes("lockedBassProbArray", (uint8_t*)arpSettings.lockedBassProbabilityValues,
+		                              RANDOMIZER_LOCK_MAX_SAVED_VALUES);
+		// Chord probability
+		writer.writeAttribute("lastLockedChordProb", arpSettings.lastLockedChordProbabilityParameterValue);
+		writer.writeAttributeHexBytes("lockedChordProbArray", (uint8_t*)arpSettings.lockedChordProbabilityValues,
+		                              RANDOMIZER_LOCK_MAX_SAVED_VALUES);
+		// Ratchet probability
+		writer.writeAttribute("lastLockedRatchetProb", arpSettings.lastLockedRatchetProbabilityParameterValue);
+		writer.writeAttributeHexBytes("lockedRatchetProbArray", (uint8_t*)arpSettings.lockedRatchetProbabilityValues,
+		                              RANDOMIZER_LOCK_MAX_SAVED_VALUES);
+		// Spread velocity
+		writer.writeAttribute("lastLockedVelocitySpread", arpSettings.lastLockedSpreadVelocityParameterValue);
+		writer.writeAttributeHexBytes("lockedVelocitySpreadArray", (uint8_t*)arpSettings.lockedSpreadVelocityValues,
+		                              RANDOMIZER_LOCK_MAX_SAVED_VALUES);
+		// Spread gate
+		writer.writeAttribute("lastLockedGateSpread", arpSettings.lastLockedSpreadGateParameterValue);
+		writer.writeAttributeHexBytes("lockedGateSpreadArray", (uint8_t*)arpSettings.lockedSpreadGateValues,
+		                              RANDOMIZER_LOCK_MAX_SAVED_VALUES);
+		// Spread octave
+		writer.writeAttribute("lastLockedOctaveSpread", arpSettings.lastLockedSpreadOctaveParameterValue);
+		writer.writeAttributeHexBytes("lockedOctaveSpreadArray", (uint8_t*)arpSettings.lockedSpreadOctaveValues,
+		                              RANDOMIZER_LOCK_MAX_SAVED_VALUES);
```

Comment:
Similar to above, these should be written via a shared implementation in `ArpeggiatorSettings`.